subcommand/mesh-init/command.go

```diff
@@ -63,31 +60,68 @@ func (c *Command) Run(args []string) int {
 }
 
 func (c *Command) realRun() error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	taskMeta, err := awsutil.ECSTaskMetadata()
 	if err != nil {
 		return err
 	}
 
 	cfg := api.DefaultConfig()
 
+	consulClient, err := api.NewClient(cfg)
+	if err != nil {
+		return fmt.Errorf("constructing consul client: %s", err)
+	}
+
+	serverConnMgrCfg, err := c.config.ConsulServerConnMgrConfig(taskMeta)
+	if err != nil {
+		return fmt.Errorf("constructing server connection manager config: %s", err)
+	}
+	serverConnMgrCfg.ServerWatchDisabled = false
+
+	watcher, err := discovery.NewWatcher(ctx, serverConnMgrCfg, c.log)
+	if err != nil {
+		return fmt.Errorf("unable to create consul server watcher: %s", err)
+	}
+
+	go watcher.Run()
+	defer watcher.Stop()
+
+	state, err := watcher.State()
+	if err != nil {
+		return fmt.Errorf("unable to fetch consul server watcher state: %s", err)
+	}
```

Comment:
non-blocking / informational: For a future PR, because the unified consul-ecs-control-plane binary will be long running in agentless, we'll need to rebuild the Consul client whenever the library switches to another Consul server.

The library has a basic [Subscribe](https://pkg.go.dev/github.com/hashicorp/consul-server-connection-manager@v0.1.2/discovery#Watcher.Subscribe) function that we can use to receive updates about Consul server addresses, as one way to do this. (Alternatively, we can periodically call `watcher.State()` and detect a server change that way which is also fine).

Reply:
Yeah. I've added a note in the PR description as well. Server watch will be taken up later on whenever we make the consul-ecs control plane a long running one.

Reply:
That's true. I think there are a few reasons for that:

1. Many consul-k8s components are short-lived (like jobs or init containers) so they use `State` once and don't need to watch for changes.
2. If gRPC is the only protocol (like in Consul Dataplane) then the gRPC connection is shared with consul-server-connection-manager and the gRPC connection automatically switches servers under the hood (no need to use State/Subscribe)
3. It's easy to use `State` to lazily construct a Consul client on-demand whenever a client is needed (consul-k8s does this in many places)

I added `Subscribe` to consul-server-connection-manager with long-running processes in mind, but it's fine to follow the consul-k8s style of using `State` everywhere (and then we could potentially remove `Subscribe` from consul-server-connection-manager it's not being used)